src/plugins/workflow_designer/src/WorkflowTabView.cpp

```diff
@@ -238,8 +238,10 @@ void WorkflowTabView::sl_dashboardsChanged(const QStringList &dashboardIds) {
         }
 
         if (!dashboardInfo.opened) {
-            dashboard->setClosed();
-            removeDashboard(dashboard);
+            if (dashboardInfo.opened != dashboard->isOpened()) {
```

Comment:
nit: I would prefer to read this check in a simpler form: `!dashboardInfo.opened && dashboard->isOpened`